README.md

```diff
@@ -4011,6 +4014,50 @@ Specifies a preshared key for the host. Valid values are 'clear', 'encrypted',
 "Specifies the preshared key password for the host. Valid value is a string.
 
 --
+### Type: cisco_upgrade
+
+Manages the upgrade of a Cisco device.
+
+| Platform | OS Minimum Version | Module Minimum Version |
+|----------|:------------------:|:----------------------:|
+| N9k      | 7.0(3)I2(1)        | 1.6.0                  |
+| N3k      | 7.0(3)I2(1)        | 1.6.0                  |
+| N5k      | not applicable     | not applicable         |
+| N6k      | not applicable     | not applicable         |
+| N7k      | not applicable     | not applicable         |
+| N9k-F    | not applicable     | not applicable         |
+
+#### <a name="cisco_upgrade-caveats">Caveats</a>
+
+The `cisco_upgrade` is only supported on *simplex* N3K and N9K devices. HA devices are currently not supported. 
+
+| Property | Caveat Description |
+|:--------|:-------------|
+| `source_uri`    | Only images on `bootflash:`  are supported. |
+
+#### Parameters

```

Comment:
Somewhere in here, it would be good to add a note indicating that the puppet file provider can be used to manage image files on bootflash: along with a pointer to the sample manifest.